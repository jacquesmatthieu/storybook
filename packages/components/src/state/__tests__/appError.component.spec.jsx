import React from 'react';
import { shallow } from 'enzyme';

import AppError from '../appError.component';

describe('AppError', () => {
  let props;

  beforeEach(() => {
    props = {};
  });

  const getWrapper = () => shallow(<AppError {...props} />);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with error details', () => {
    // Given
    props.error = new Error('Foo');
    props.error.stack = 'Awesome stack trace';

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
