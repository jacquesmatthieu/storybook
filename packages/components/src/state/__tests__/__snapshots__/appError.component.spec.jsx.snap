// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppError should render properly 1`] = `
<appErrorcomponent__ErrorWrapper>
  <typographycomponent__Typography
    variant="headline"
  >
    error.title
  </typographycomponent__Typography>
  <br />
  <typographycomponent__Typography
    variant="body"
  >
    error.appError
  </typographycomponent__Typography>
  <br />
  <br />
  <Button
    color="primary"
    hollow={true}
    onClick={[Function]}
    type="button"
  >
    reload
  </Button>
</appErrorcomponent__ErrorWrapper>
`;

exports[`AppError should render with error details 1`] = `
<appErrorcomponent__ErrorWrapper>
  <typographycomponent__Typography
    variant="headline"
  >
    error.title
  </typographycomponent__Typography>
  <br />
  <typographycomponent__Typography
    variant="body"
  >
    error.appError
  </typographycomponent__Typography>
  <br />
  <appErrorcomponent__ErrorCodeBlock>
    Awesome stack trace
  </appErrorcomponent__ErrorCodeBlock>
  <br />
  <Button
    color="primary"
    hollow={true}
    onClick={[Function]}
    type="button"
  >
    reload
  </Button>
</appErrorcomponent__ErrorWrapper>
`;
